week3/Q1.py

```diff
@@ -0,0 +1,177 @@
+def get_graph(values, n, m):
+	'''
+	Builds a graph.
+
+	Args:
+		values: str, elements of the table, all rows in one line
+		n: int, number of columns
+		m: int, number of rows
+
+	Returns:
+		dict, includes two elements with keys 'nodes' (stores a dict where key is number of a node and value is node's value) and 'vertices' (stores a dict where key is a number of a node and value is a set of numbers of nodes to which it is connected)
+	'''
+	nodes = {i: values[i] for i in range(n*m)}
+	vertices = []
+	for i in range(n*m):
+		if n == 1 and m == 1:
+			vertices.append({})
+			break
+		elif n == 1 or m == 1:
+			if i == 0:
+				vertices.append({i + 1})
+			elif (i == m - 1 and n == 1) or (i == n - 1 and m == 1):
+				vertices.append({i - 1})
+			else:
+				vertices.append({i - 1, i + 1})
+		else:
+			#up left corner
```

Comment:
Some further hints to unblock you here.

Instead of explicitly handling the various cases, let's consider the most generic case:

At location `i`: the neighbours are:
* `i - 1` to the left
* `i + 1` to the right
* `i - n` above
* `i + n` below
* `i -n - 1` above to the left
* `i - n + 1`above to the right
* `i + n - 1` below to the left
* `i + n + 1` below to the right

One way to think of this is: if we consider `(r, c)` row and column coordinates instead of `i`, but we know that `i = r * n + c`. If we change `r` and `c` to move, we can check whether the combination is possible by check that `r` and `c` are within the proper boundaries, i.e., `0 <= r <= m and 0 <= c <= n`.

With that we should be able to enumerate all the options without having to list all the cases.

Let me know if this is clear enough.